ChatApp/Presentation/View/SwitchView/SwitchModeView.swift

```diff
@@ -0,0 +1,38 @@
+import UIKit
+
+// MARK: Switch Protocol
+protocol SwitchModeViewDelegate: AnyObject {
+    func switchModeView(with state: SwitchModeView.ButtonState)
```

Comment:
try matching Apple's style of writing a delegate. In that case it would be:

`func switchModeView(_ switchModeView: SwitchModeView, didSwitchStateTo state: SwitchModeView.ButtonState)`